### ISSUE TITLE
Convergence error while using Pdep: "itmax iterations solving bounded least squares problem"

### ISSUE THREAD

**Issue by alongd** [ReactionMechanismGenerator/RMG-Py#544]
Throughout all my Pdep runs so far I get repeated convergence errors at the end of the job.log file such as:

```
MODEL GENERATION COMPLETED

The final model core has 248 species and 4185 reactions
The final model edge has 4744 species and 11144 reactions

RMG execution terminated at Tue Jan  5 18:05:58 2016
 error number =    22, message level =     1
 i1 =      125
 dbolsm(). more than (i1)=itmax iterations solving bounded least squares problem.
 error number =    22, message level =     1
 i1 =      120
 dbolsm(). more than (i1)=itmax iterations solving bounded least squares problem.
 error number =    22, message level =     1
 i1 =      125
 dbolsm(). more than (i1)=itmax iterations solving bounded least squares problem.
 error number =    22, message level =     1
 i1 =      125
 dbolsm(). more than (i1)=itmax iterations solving bounded least squares problem.
 error number =    22, message level =     1
 i1 =      125
```

and so on.
The run does reach completion, but it would be nice to get rid of the error messages.
Any ideas?

**Comment by rwest** [comment]
I expect they're coming from DQED via PyDQED, so you could start by looking there.

**Comment by soumsrani** [comment]
Hi, I used RMG ver 1.0.4. I am having similar error as above, but in my case the model is not converged. The error is shown below. The input and stdout.txt files are attached. Any suggestions as to how to resolve this error? Thank you.

[stdout.txt](https://github.com/ReactionMechanismGenerator/RMG-Py/files/260265/stdout.txt)

[input.txt](https://github.com/ReactionMechanismGenerator/RMG-Py/files/260271/input.txt)


Conducting simulation of reaction system 5...
At time 2.9579e-02 s, species [C]#CC#C(2810) exceeded the minimum rate for moving to model core
    Characteristic rate: 2.5901e+00 mol/m^3*s
    [C]#CC#C(2810) rate: 1.3044e+00 mol/m^3*s (0.5036)
    PDepNetwork #1391 leak rate: 1.3010e+00 mol/m^3*s (0.5023)
At time 5.0000e-02 s, reached target termination time.

Traceback (most recent call last):
  File "/home/soumyag/workspace/RMG-Py/rmg.py", line 165, in <module>
    rmg.execute(inputFile, output_dir, **kwargs)
  File "/home/soumyag/workspace/RMG-Py/rmgpy/rmg/main.py", line 593, in execute
    self.reactionModel.prune(self.reactionSystems, self.fluxToleranceKeepInEdge, self.maximumEdgeSpecies, self.minSpeciesExistIterationsForPrune)
  File "/home/soumyag/workspace/RMG-Py/rmgpy/rmg/model.py", line 1133, in prune
    index = self.edge.species.index(spec)
ValueError: Species(index=2798, label="[CH]=[C]C#C", thermo=NASA(polynomials=[NASAPolynomial(coeffs=[2.53264,0.0269958,-3.04872e-05,1.76606e-08,-3.72335e-12,74012.9,13.51], Tmin=(100,'K'), Tmax=(1422.33,'K')), NASAPolynomial(coeffs=[8.18732,0.00427854,6.57397e-07,-3.05879e-10,2.66597e-14,73093.7,-13.3395], Tmin=(1422.33,'K'), Tmax=(5000,'K'))], Tmin=(100,'K'), Tmax=(5000,'K'), E0=(614.901,'kJ/mol'), comment="""Thermo library: DFT_QCI_thermo + radical(C=C=CJ) + radical(C=C=CJ)"""), conformer=Conformer(E0=(614.901,'kJ/mol'), modes=[HarmonicOscillator(frequencies=([540,563.333,586.667,610,1970,2140,182.461,249.239,821.659,2007.63,2008.53,2009.86],'cm^-1'))], spinMultiplicity=3), molecule=[Molecule(SMILES="[CH]=C=C=[CH]"), Molecule(SMILES="[CH]=[C]C#C")], transportData=TransportData(shapeIndex=2, epsilon=(3010.21,'J/mol'), sigma=(5.08147,'angstroms'), dipoleMoment=(0,'C*m'), polarizability=(0,'angstroms^3'), rotrelaxcollnum=0, comment="""Epsilon & sigma estimated with Tc=470.19 K, Pc=52.06 bar (from Joback method)"""), molecularWeight=(50.0587,'amu'), energyTransferModel=SingleExponentialDown(alpha0=(3.5886,'kJ/mol'), T0=(300,'K'), n=0.85)) is not in list
 error number =    22, message level =     1
 i1 =      110
 dbolsm(). more than (i1)=itmax iterations solving bounded least squares problem.
 error number =    22, message level =     1
 i1 =      115
 dbolsm(). more than (i1)=itmax iterations solving bounded least squares problem.
 error number =    22, message level =     1
 i1 =      110
 dbolsm(). more than (i1)=itmax iterations solving bounded least squares problem.
 error number =    22, message level =     1
 i1 =       95
 dbolsm(). more than (i1)=itmax iterations solving bounded least squares problem.
 error number =    22, message level =     1
 i1 =      115
 dbolsm(). more than (i1)=itmax iterations solving bounded least squares problem.
 error number =    22, message level =     1
 i1 =      120

**Comment by rwest** [comment]
Hi @soumsrani,
the warnings about the maximum iterations when solving the bounded least squares problem are only warnings, it's just that they all get held in a buffer until the job finishes then suddenly dumped to stdout (something we should fix, and relevant to this issue #544).

What actually caused your job to crash is in the stack trace that appears just before that:
```python
Traceback (most recent call last):
  File "/home/soumyag/workspace/RMG-Py/rmg.py", line 165, in <module>
    rmg.execute(inputFile, output_dir, **kwargs)
  File "/home/soumyag/workspace/RMG-Py/rmgpy/rmg/main.py", line 593, in execute
    self.reactionModel.prune(self.reactionSystems, self.fluxToleranceKeepInEdge, self.maximumEdgeSpecies, self.minSpeciesExistIterationsForPrune)
  File "/home/soumyag/workspace/RMG-Py/rmgpy/rmg/model.py", line 1133, in prune
    index = self.edge.species.index(spec)
ValueError: Species(index=2798, label="[CH]=[C]C#C", etc.) is not in list
```
and seems to be a problem with pruning, where a certain species is not in the `self.edge.species` list.
Looks like issue #600. Please repost this there.
